src/woo_publications/metadata/tests/test_informatie_categorie_admin.py

```diff
@@ -0,0 +1,212 @@
+                self.assertInHTML(
+                    "https://www.example.com/waardenlijsten/2",
+                    str(response.content),
+                )
+
+    def test_informatie_category_admin_can_not_update_item_with_oorsprong_waardenlijst(
+        self,
+    ):
+        url = reverse(
+            "admin:metadata_informatiecategorie_change",
+            kwargs={"object_id": self.information_category_value_list.id},
+        )
+
+        response = self.app.get(url, user=self.user)
+        form = response.forms["informatiecategorie_form"]
+
+        self.assertEqual(response.status_code, 200)
+        # page doesn't have save button.
+        with self.assertRaisesMessage(
+            AssertionError,
+            'Couldn\'t find \'<input class="default" name="_save" type="submit" value="Opslaan">\' in response',
+        ):
+            self.assertInHTML(
+                '<input type="submit" value="Opslaan" class="default" name="_save">',
+                str(response.content),
+            )
+
+        with self.assertRaisesMessage(
+            AssertionError,
+            "No field by the name 'identifier' found (fields: 'csrfmiddlewaretoken')",
+        ):
+            form["identifier"] = "https://www.example.com/waardenlijsten/12"
+
+    def test_informatie_category_admin_can_update_item_with_oorsprong_zelf_toegevoegd(
+        self,
+    ):
+        url = reverse(
+            "admin:metadata_informatiecategorie_change",
+            kwargs={"object_id": self.information_category_custom_entry.id},
+        )
+
+        response = self.app.get(url, user=self.user)
+        form = response.forms["informatiecategorie_form"]
+
+        self.assertEqual(response.status_code, 200)
+        # page has save button.
+        self.assertInHTML(
+            '<input type="submit" value="Opslaan" class="default" name="_save">',
+            str(response.content),
+        )
+
+        # test if identifier isn't editable
+        with self.assertRaises(AssertionError):
+            form["identifier"] = "https://www.example.com/waardenlijsten/12"
+
+        form["naam"] = "changed"
+        form["naam_meervoud"] = "changed"
+        form["definitie"] = "changed"
+        response = form.submit("submit")
+
+        self.assertEqual(response.status_code, 302)
+        self.information_category_custom_entry.refresh_from_db()
+
+        self.assertEqual(self.information_category_custom_entry.naam, "changed")
+        self.assertEqual(
+            self.information_category_custom_entry.naam_meervoud, "changed"
+        )
+        self.assertEqual(self.information_category_custom_entry.definitie, "changed")
+
+    def test_informatie_category_admin_create_item(self):
+        response = self.app.get(
+            reverse("admin:metadata_informatiecategorie_add"), user=self.user
+        )
+        form = response.forms["informatiecategorie_form"]
+
+        form["identifier"] = "https://www.example.com/waardenlijsten/999"
```

Comment:
this is not something that users should be entering themselves, we need to generate it